assets/js/base/hooks/shipping/use-shipping-address.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useReducer } from '@wordpress/element';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { useDebounce } from 'use-debounce';
+import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+
+/**
+ * Internal dependencies
+ */
+import { useStoreCart } from '../cart/use-store-cart';
+import { pluckAddress } from '../../utils';
+
+const reducer = ( state, address ) =>
+	isShallowEqual( pluckAddress( address ), pluckAddress( state ) )
+		? state
+		: { ...state, ...address };
+
+export const useShippingAddress = () => {
+	const { shippingAddress: initialAddress } = useStoreCart();
+	const [ shippingAddress, setShippingAddress ] = useReducer(
+		reducer,
+		initialAddress
```

Comment:
now that we receive shipping address schema from the backend, we don’t need to pass it down. I love this, I didn’t like the previous solution I implemented.